### ISSUE TITLE
Snap buil of version 3.1.6.1 fail

### ISSUE THREAD

**Issue by nicolargo** [nicolargo/glances#1801]
The last snap build failed ==> https://snapcraft.io/glances/builds/1270664#footer

@Lin-Buo-Ren Can you have a look on it ?

**Comment by nicolargo** [comment]
Ok i understand the reason why it fail... 

Pip 21.0 do not support Python 2 anymore...

**Comment by nicolargo** [comment]
Change done in commit 102a6fb9 to force the use of Ubuntu 20.04 as base.

**Comment by nicolargo** [comment]
Better but not perfect...
==> https://snapcraft.io/glances/builds/1270820

@Lin-Buo-Ren Any idea ?

**Comment by nicolargo** [comment]
Another build another issue...

Snap is pain in the ass.

**Comment by akrea** [comment]
Hi

I posted my problem and solving attempts on [#1738 ](https://github.com/nicolargo/glances/issues/1738). Only after that I noticed this thread.

**Comment by akrea** [comment]
BTW: THANKS FOR YOUR AWESOME WORK!!!

**Comment by nicolargo** [comment]
Last build error...


```
error: cannot pack "/build/glances/prime": cannot validate snap "glances": invalid definition of application "glances": app description field 'command' contains illegal "bin/glances-launch $SNAP/bin/glances\n" (legal: '^[A-Za-z0-9/. _#:$-]*$')
```

@Lin-Buo-Ren Any idea ?

**Comment by nicolargo** [comment]
I change the command to:

```
command: desktop-launch $SNAP/bin/glances
```

Another issue: https://snapcraft.io/glances/builds/1331949

Perhaps the file should be refactor from scratch.

See => https://github.com/videolan/vlc/blob/master/extras/package/snap/snapcraft.yaml

**Comment by nicolargo** [comment]
Perhaps should have a look o a working package:
- https://git.launchpad.net/~chromium-team/chromium-browser/+git/snap-from-source/tree/snapcraft.yaml?h=beta
-

**Comment by nicolargo** [comment]
Corrected by last commit
